scripts/base/protocols/ldap/main.zeek

```diff
@@ -186,75 +187,107 @@ event LDAP::message(c: connection,
                     object: string,
                     argument: string) {
 
+  # TODO: Should this handle Unsolicited Notification (message_id == 0) and
+  #       log it directly?
+  #       See: https://datatracker.ietf.org/doc/html/rfc4511#section-4.4.
+
   if (opcode == LDAP::ProtocolOpcode_SEARCH_RESULT_DONE) {
     set_session(c, message_id, opcode);
 
-    local searches = c$ldap$searches[message_id];
+    local sm = c$ldap$searches[message_id];
 
     if ( result != LDAP::ResultCode_Undef ) {
-      if ( ! searches?$results )
-        searches$results = set();
-      add searches$results[RESULT_CODES[result]];
+      local sresult_str = RESULT_CODES[result];
+      if ( sm?$result && sm$result != sresult_str ) {
+        Reporter::conn_weird("LDAP_search_result_change", c,
+                             fmt("%s: %s -> %s", message_id, sm$result, sresult_str), "LDAP");
+        sm$result = cat_sep("|", "", sm$result, sresult_str);
```

Comment:
Hmm, after writing the NEWS entry, I'm now strongly leaning towards just overwrite the existing value. The concatenation approach directly translates into unbounded state growth and potentially the creation of immense log entries (something which the existing logic suffers from as well).